### ISSUE TITLE
[Bug] Custom ch param in YAML is overwritten by default ch=3 in DetectionModel initialization (YOLO11-OBB)

### ISSUE THREAD

**Issue by songxiuqiang** [ultralytics/ultralytics#23083]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I encountered an issue when trying to build a custom multi-channel **YOLO11-OBB** model (e.g., 6 channels).

Even though I explicitly defined `ch: 6` in my custom YAML configuration file (e.g., `yolo11n-obb-6ch.yaml`), the model is always initialized with `ch=3`. This causes a shape mismatch error during training because the first layer is built for 3 (or 4) channels instead of the specified 6.

### **Root Cause Analysis**
The issue lies in the `__init__` method of the `DetectionModel` class (parent of `OBBModel`).

The `__init__` method has a default argument `ch=3`. After loading the YAML configuration into `self.yaml`, the code immediately assigns the argument `ch` to `self.yaml["channels"]`, ignoring whatever value was present in the loaded YAML dictionary.

**Current Code logic (Conceptual):**
```python
class DetectionModel(BaseModel):
    def __init__(self, cfg="yolo11n.yaml", ch=3, nc=None, verbose=True):
        # ...
        self.yaml = cfg if isinstance(cfg, dict) else yaml_model_load(cfg)
        
        # PROBLEM HERE:
        # The argument 'ch' (defaulting to 3) overwrites the 'ch' defined in the YAML file.
        self.yaml["channels"] = ch 
        
        # ...
        self.model, self.save = parse_model(deepcopy(self.yaml), ch=ch, verbose=verbose)
```

### **To Reproduce**
1. Create a custom YAML file (e.g., `custom_obb.yaml`) and set `ch: 6`.
2. Initialize the model using the YAML file directly:
   ```python
   model = YOLO("custom_obb.yaml") # Arguments ch default to 3 here internally
   ```
3. Check the input channels of the model; it will be 3 instead of 6.

### **Proposed Fix**
We should prioritize the `ch` value defined in the YAML file if it exists.

I was able to fix this locally by adding a check before the assignment.

**In `ultralytics/nn/tasks.py` (inside `DetectionModel.__init__`):**

```python
        # ... (after deprecated Silence module check) ...

        # Fix: Retrieve 'ch' from YAML if it exists, otherwise use the argument default
        ch = self.yaml.get("ch", ch) 
        
        self.yaml["channels"] = ch  # save channels
        
        if nc and nc != self.yaml["nc"]:
        # ...
```

By adding `ch = self.yaml.get("ch", ch)`, the configuration correctly respects the user-defined channel count in the YAML file.

---

### Environment

Ultralytics 8.3.159 🚀 Python-3.10.14 torch-2.9.1+cu128 CUDA:0 

OS                  Linux-6.8.0-87-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.14
Install             git
GPU count           1
CUDA                12.8

numpy               ✅ 1.26.4>=1.23.0
matplotlib          ✅ 3.8.0>=3.3.0
opencv-python       ✅ 4.11.0.86>=4.6.0
pillow              ✅ 11.3.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.5>=2.23.0
scipy               ✅ 1.11.3>=1.4.1
torch               ✅ 2.9.1>=1.8.0
torch               ✅ 2.9.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.24.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 7.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.3.1>=1.1.4
ultralytics-thop    ✅ 2.0.18>=2.0.0

### Minimal Reproducible Example

```python
class DetectionModel(BaseModel):
    def __init__(self, cfg="yolo11n.yaml", ch=3, nc=None, verbose=True):
        # ...
        self.yaml = cfg if isinstance(cfg, dict) else yaml_model_load(cfg)
        
        # PROBLEM HERE:
        # The argument 'ch' (defaulting to 3) overwrites the 'ch' defined in the YAML file.
        self.yaml["channels"] = ch 
        
        # ...
        self.model, self.save = parse_model(deepcopy(self.yaml), ch=ch, verbose=verbose)
```

``` yaml
# Parameters in cfg
ch: 6
nc: 80 # number of classes
scales: # model compound scaling constants, i.e. 'model=yolo11n-obb.yaml' will call yolo11-obb.yaml with scale 'n'
  # [depth, width, max_channels]
  n: [0.50, 0.25, 1024] # summary: 196 layers, 2695747 parameters, 2695731 gradients, 6.9 GFLOPs
  s: [0.50, 0.50, 1024] # summary: 196 layers, 9744931 parameters, 9744915 gradients, 22.7 GFLOPs
  m: [0.50, 1.00, 512] # summary: 246 layers, 20963523 parameters, 20963507 gradients, 72.2 GFLOPs
  l: [1.00, 1.00, 512] # summary: 372 layers, 26220995 parameters, 26220979 gradients, 91.3 GFLOPs
  x: [1.00, 1.50, 512] # summary: 372 layers, 58875331 parameters, 58875315 gradients, 204.3 GFLOPs

 ...
```

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @songxiuqiang, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for the detailed report and root-cause analysis around custom multi-channel **YOLO11-OBB** YAML `ch` handling—this is helpful context 🧩

This is an automated response 🤖 to help triage quickly, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. You already included a good snippet; to make it actionable for maintainers, please also include a runnable MRE that shows the behavior end-to-end (e.g., minimal YAML + the exact `YOLO("custom_obb.yaml")` call + the full stack trace / model summary showing the channel mismatch) 🔎

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
It's not a bug because `ch` inside the model YAML is not a valid parameter. It existed in the YOLOv5 repo, but it's not a valid parameter in Ultralytics.

The actual channel shape used during training depends on the dataset YAML and not the model YAML.

**Comment by songxiuqiang** [comment]
Thanks @Y-T-G, I have tried your solution and it works.
